tests/test_templates.py

```diff
@@ -12,106 +12,307 @@
+            return [qml.expval(qml.Identity(0)), qml.expval(qml.X(1))]
+
+        circuit(**inpts)
+
+
+class TestIntegrationCircuitTf:
+    """Tests the integration of templates into circuits using the TensorFlow interface."""
+
+    @pytest.mark.parametrize(fixture, qubit_const)
+    def test_integration_qubit_positional_arg(self, template, inpts):
+        """Checks integration of qubit templates using positional arguments to qnode."""
+
+        inpts = [tf.Variable(i) for i in inpts]
+        dev = qml.device('default.qubit', wires=2)
+
+        @qml.qnode(dev, interface='tf')
+        def circuit(*inp_):
+            qml.PauliX(wires=0)
+            template(*inp_, wires=range(2))
+            template(*inp_, wires=range(2))
+            qml.PauliX(wires=1)
+            return [qml.expval(qml.Identity(0)), qml.expval(qml.PauliX(1))]
+
+        circuit(*inpts)
+
+    @pytest.mark.parametrize(fixture, qubit_const)
+    def test_integration_qubit_keyword_arg(self, template, inpts):
+        """Checks integration of qubit templates using keyword arguments to qnode."""
+
+        inpts = {"w"+str(1): tf.Variable(inpts[i]) for i in range(len(inpts))}
+        dev = qml.device('default.qubit', wires=2)
+
+        @qml.qnode(dev, interface='tf')
+        def circuit(**inp_):
+            qml.PauliX(wires=0)
+            template(*inp_.values(), wires=range(2))
+            template(*inp_.values(), wires=range(2))
+            qml.PauliX(wires=1)
+            return [qml.expval(qml.Identity(0)), qml.expval(qml.PauliX(1))]
+
+        circuit(**inpts)
+
+    @pytest.mark.parametrize(fixture, cv_const)
+    def test_integration_cv_positional_args(self, gaussian_device_2_wires, template, inpts):
+        """Checks integration of continuous-variable templates using positional arguments to qnode."""
+
+        inpts = [tf.Variable(i) for i in inpts]
+
+        @qml.qnode(gaussian_device_2_wires, interface='tf')
+        def circuit(*inp_):
+            qml.Displacement(1., 1., wires=0)
+            template(*inp_, wires=range(2))
+            template(*inp_, wires=range(2))
+            qml.Displacement(1., 1., wires=1)
+            return [qml.expval(qml.Identity(0)), qml.expval(qml.X(1))]
+
+        circuit(*inpts)
+
+    @pytest.mark.parametrize(fixture, cv_const)
+    def test_integration_cv_keyword_args(self, gaussian_device_2_wires, template, inpts):
+        """Checks integration of continuous-variable templates using keyword arguments to qnode."""
+
+        inpts = {"w"+str(i): tf.Variable(inpts[i]) for i in range(len(inpts))}
+
+        @qml.qnode(gaussian_device_2_wires, interface='tf')
+        def circuit(**inp_):
+            qml.Displacement(1., 1., wires=0)
+            template(*inp_.values(), wires=range(2))
+            template(*inp_.values(), wires=range(2))
+            qml.Displacement(1., 1., wires=1)
+            return [qml.expval(qml.Identity(0)), qml.expval(qml.X(1))]
 
-        circuit(weights=p)
+        circuit(**inpts)
```

Comment:
Since the TF and Torch tests largely duplicate the numpy tests, can we parameterize them and just have one test class?

Reply:
Yes, if we turn the qnodes themselves into fixtures. I considered this, but it obscures the tests really a lot, and is hard to know what is even tested...

Reply:
Also, one may have templates that need a separate test, which can then be easily added to the class. I'm in favour of repeating code here, for ease of use.

Reply:
I reconsidered this, and now parametrized the interfaces as well. Works like a charm.